### ISSUE TITLE
Subfolder structure of sketch lost after "Save As" operation

### ISSUE THREAD

**Issue by jsmwrench** [arduino/arduino-ide#2077]
### Describe the problem

Arduino sketches may contain files under subfolders:

- [`data`](https://arduino.github.io/arduino-cli/dev/sketch-specification/#data-subfolder)
- [`src`](https://arduino.github.io/arduino-cli/dev/sketch-specification/#src-subfolder)

🐛 After a "**Save As...**" operation, any subfolders are lost and the files are all moved to the root sketch folder.

### To reproduce

1. Click the following link to download the demonstration sketch:
   [HasSubfolders.zip](https://github.com/arduino/arduino-ide/files/12036173/HasSubfolders.zip)
1. Unzip the downloaded file.
1. Take note of the file structure of the sketch:
   ```text
   HasSubfolders/
   ├── HasSubfolders.ino
   └── src/
       └── FromSubfolder.h
   ```
1. Select **File > Open** from the Arduino IDE menus.
1. Select the `HasSubfolders.ino` file from the unzipped folder.
1. Select **File > Save As...**" from the Arduino IDE menus.
1. Save the sketch to any convenient name and location.
1. Select **Sketch > Show Sketch Folder** from the Arduino IDE menus.

🐛 The subfolder structure of the sketch was lost:

```text
HasSubfoldersAfterSaveAs/
├── FromSubfolder.h
└── HasSubfoldersAfterSaveAs.ino
```

### Expected behavior

A "**Save As**" operation does not make any changes to the sketch file structure.

### Arduino IDE version

94d2962

### Operating system

- Windows
- Linux
- macOS

### Operating system version

- Windows 11
- Ubuntu 22.04
- macOS Ventura

### Additional context

I bisected the regression to 0ab2826 (the fault does not occur when using 2aad0e3).

#### Additional reports

- https://github.com/arduino/arduino-ide/issues/2077#issuecomment-1567372310
- https://forum.arduino.cc/t/ide-2-2-1-save-as-does-not-copy-subfolders-correctly/1191437

### Issue checklist

- [X] I searched for previous reports in [the issue tracker](https://github.com/arduino/arduino-ide/issues?q=)
- [X] I verified the problem still occurs when using the latest [nightly build](https://www.arduino.cc/en/software#nightly-builds)
- [X] My report contains all necessary details

**Comment by per1234** [comment]
Hi @jsmwrench. Thanks for your report.

>the original file was opened from an external drive. I then used File/Save As to save it to my "sketches" folder.

Please check to see whether the unexpected 36 files are also present in the folder on that external drive.

A common misconception is that an Arduino sketch is a file. The sketch is actually a folder. When you do a "Save As" operation with a sketch, all the files from the original sketch folder are saved to the folder that is the target of the "Save As" operation. So if you had previously placed all those 36 files in the original sketch folder then it is expected that they would also be present in the new folder as well.

**Comment by fwlrichard** [comment]
Save as is also dumping All Files from all sub folders into the Sketch folder.

Original Sketch Folder Containing only the Sketch and a data folder. Data folder contains 4xfile.txt files.

![SaveAs_Issue_ORG](https://github.com/arduino/arduino-ide/assets/134958880/2a9d1e9c-c646-49b4-ae34-3c6839ad119a)

Open Sketch click "Save As" and as you can see now the new Sketch folder contains the sketch and all 4 of the data files from the "data" folder that is now gone.

![SaveAs_Issue_BAD](https://github.com/arduino/arduino-ide/assets/134958880/7d028080-b32d-409f-bb26-2477734bb5fc)

@per1234 - All previous versions of the Arduino IDE have been able to maintain file structure when doing a "Save As. " 
Problem started with IDE Version: 2.1.0
Date: 2023-04-19T15:31:10.185Z
CLI Version: 0.32.2

**Comment by jsmwrench** [comment]
None of the new files are in the original folder.  The original folder only had the sketch itself and the AccelStepper library in it.  I opened the sketch in VS Code and saved as a new file on my computer and when I did, the IDE created all of these files in the same folder as the newly saved sketch.
Folder I took it from:
![From folder](https://github.com/arduino/arduino-ide/assets/14171170/0a625322-a51f-42d2-b858-6555b4f76a27)

New Folder:
![New Folder](https://github.com/arduino/arduino-ide/assets/14171170/f929e127-e791-4577-bb6b-7aeebe5a9cfe)
